### ISSUE TITLE
Unable to retrieve child resources

### ISSUE THREAD

**Issue by yThomasS** [Microsoft/AzureStorageExplorer#1152]
**Storage Explorer Version**: 1.6.2
**Platform/OS**: Windows Server 2016
**Architecture**: x64
**Regression From**: N/A

**Bug description**
Try to connect to Blob Storage, but only get error "Unable to retrieve child resources. Details: This request is not authorized to perform this operation.

**Steps to Reproduce**
1. Start Azure Storage Explorer
2. Click on Blob Container

**Expected Experience**
Should show Blob containers

**Actual Experience**
Shows error message

**Additional Context**
It is working from a Mac, so configuration seems to be ok.
The Windows Server is hosted in Azure and has a public IP address.
<img width="1472" alt="screen shot 2019-02-21 at 15 36 37" src="https://user-images.githubusercontent.com/37152106/53177211-e52e8d80-35ef-11e9-8eec-20fd0c8392c2.png">

**Comment by yThomasS** [comment]
Is maybe a problem with the IP address. If I create a connection with a general Access Key the connection is ok. With a SAS without IP address it is ok. But if I specify the IP address of the Windows Server in the SAS it does not work any more.

**Comment by mikecasas** [comment]
Getting the same error.

I deleted the folder as described in https://github.com/Microsoft/AzureStorageExplorer/issues/436#issuecomment-461548078
but it didn't fix the issue.

![1](https://user-images.githubusercontent.com/11261164/53341826-96784080-38da-11e9-94d3-a479b421d776.PNG)
![2](https://user-images.githubusercontent.com/11261164/53341831-98da9a80-38da-11e9-9aa7-b684cae3f112.PNG)

**Comment by mikecasas** [comment]
When I used to the Cloud Explorer from within Visual Studio, I got a slightly different error message. It said it was not initialized and I had to run the init command.

I opened a command line and navigated to 
``` C:\Program Files (x86)\Microsoft SDKs\Azure\Storage Emulator```

then i ran
```
AzureStorageEmulator init
```
That did the trick and I am able to see my local Tables from Cloud Explorer within VS2017 and from the MSFT Azure Storage Explorer.

The doc below provided the basic guidance.

https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator

**Comment by LavMehta** [comment]
![errorn](https://user-images.githubusercontent.com/45895526/53812879-a805bc00-3f5c-11e9-8cf3-790f22f16742.JPG)
Same error

**Comment by craxal** [comment]
Hi everyone! We're happy to help you get to the bottom of this, but we need you to give us more information about what you're trying to accomplish. Please make sure you're providing steps on what you're clicking, what you're trying to connect to, and so on.

@yThomasS It appears you're attempt to connect using a SAS. Can you tell us where you got the SAS URI from? What permissions does it have? If possible please share the URI you have, redacting the signature portion of the SAS token.

@mikecasas Were you attempting to connect to a storage emulator? If so, note that Storage Explorer does not start the emulator automatically (see [Getting Started](https://docs.microsoft.com/en-us/azure/vs-azure-tools-storage-manage-with-storage-explorer?toc=%2Fazure%2Fstorage%2Ffiles%2Ftoc.json&tabs=windows#work-with-local-development-storage) page).

@LavMehta We need more context on what you're trying to connect to. Please give us some steps that we can easily follow.

**Comment by LavMehta** [comment]
Hi @craxal ; I installed Azure storage explorer and then by clicking on the connector button on left , i started connecting the azure storage using a connection string, as soon as the connection was successful, i clicked on the storage account and , then on Blob Containers . After that it keeps on loading , and neither i can access anything inside my blob container nor i can upload any data. And after some time i get the error i showed above. 
I tried  making connections using Key as well. 
In all the cases connection is scuccessful but still there is no access. 
i hope it makes picture a bit clear to you. 
Thanks in Advance for helping.

**Comment by mikecasas** [comment]
@craxal, Yes it was the emulator. I figured that out from the docs, but just left the notes for anyone else; or even for my future self.  Thanks!

**Comment by pflickin** [comment]
Having the same problem on osx.  I reset storage explorer, added my credentials, expand one of my subscriptions, then get:

Unable to retrieve child resources.
Details: Cannot read property 'connectionString' of undefined

Please advise.  This makes storage explorer unusable.

**Comment by craxal** [comment]
@LavMehta Click on the account node and look at the Properties tab below. Do you see service endpoints there? Do they appear correct? Also, is it possible to access your storage accounts by signing in instead?

**Comment by MRayermannMSFT** [comment]
Closing due to inactivity from original author. If you're still facing issues please open a new issue. Thanks!

**Comment by MRayermannMSFT** [comment]
Closing due to inactivity from original author. If you're still facing issues please open a new issue. Thanks!

**Comment by LavMehta** [comment]
Thanks craxal and extremely sorry for the late reply, my issue got solved on it's own after reinstalling the Storage explorer.

**Comment by NoelAbrahams** [comment]
In my case this the connection to CosmosDB was blocked by the firewall. The old DTUI actually provided me with an informative error message:

![image](https://user-images.githubusercontent.com/1106823/74825969-b3373080-5302-11ea-9c61-133409cab00c.png)


Perhaps you guys can learn something about UX from that?
